### PR TITLE
[Core] Remove capture the CarouselView Experimental flag exception

### LINKED ISSUE #11236: [Bug] No Flags needed for CarouselView
### Description

I am able to use CarouselView without the application throwing an InvalidOperationException. Looks like this [merge](https://github.com/xamarin/Xamarin.Forms/commit/87d720c71ec1c8f8fea2daf2105ba4b6689d71f1#diff-a838fddaa0fe09f722bf5328649e51ac) removed the need for the flag as a result of this [PR](https://github.com/xamarin/Xamarin.Forms/pull/7465/files) however [MS documentation](https://docs.microsoft.com/en-us/xamarin/xamarin-forms/user-interface/carouselview/) still lists the control as being in preview.

### Steps to Reproduce

1. Create a forms project and add a carouselview, without adding any forms flags.

### Expected Behavior

1. An InvalidOperationException is thrown when running the application. 

### Actual Behavior

User can use the CarouselView

### Basic Information

- Version with issue: 4.4+ I think
- Last known good version: -

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change ###

Remove capture the CarouselView Experimental flag exception.

### Issues Resolved ###

- fixes #11236

### API Changes ###
 
 None

### Platforms Affected ### 

- Core/XAML (all platforms)

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

Not applicable

### Testing Procedure ###
Launch Core Gallery and navigate to the CarouselView Gallery. Try to navigate to any sample without setting the "CarouselView_Experimental" flag. An exception must be thrown.

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Xamarin.Forms.Core/Items/CarouselView.cs [hunk 1]
```diff
@@ -201,15 +201,8 @@ public CarouselView()
...
-			try
-			{
-				ExperimentalFlags.VerifyFlagEnabled(nameof(CollectionView), ExperimentalFlags.CarouselViewExperimental,
-					constructorHint, memberName);
-			}
-			catch (InvalidOperationException)
-			{
-
-			}
+			ExperimentalFlags.VerifyFlagEnabled(nameof(CollectionView), ExperimentalFlags.CarouselViewExperimental,
```
Comment (samhouts) [inline Xamarin.Forms.Core/Items/CarouselView.cs:204]: nameof CarouselView?

Comment (jsuarezruiz) [reply Xamarin.Forms.Core/Items/CarouselView.cs]: Changed!

**Comment by jsuarezruiz** [comment]
@rmarinho Fixed the failing UITests.

**Comment by samhouts** [comment]
I think it's a bit late to do this.

**Pull request closed without merging**